org.eclipse.ice.modeling/src/main/java/org/eclipse/ice/modeling/workflowEngine/Procedure.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.ice.modeling.workflowEngine;
+
+/**
+ * The Procedure hold the set of steps that constitute the workflow.  
+ * My current thinking is that this class does not hold state, only Steps
+ * and Completion Criteria
+ */
+public class Procedure {
+
+	private Step[] steps;    // This should really be a List where steps can be added dynamically
+	private CompleteCriteria completeCriteria;
+	private int curIndex;
+	private int numSteps;
+
+	public Procedure() {
+		System.out.println("Procedure() constructor");
+		
+		// Create a default set of steps
+		// This should really be a List where steps can be added dynamically
+		numSteps = 3;
+		steps = new Step[numSteps];
+		System.out.println( "##### BEGIN:  CREATE SETPS FOR PROCEDURE " );
+		for (int i=0; i < 3; i++) {
```

Comment:
I think we shouldn't have this in the constructor at all, and just expect steps to be added dynamically. That way there is no confusion about what a procedure actually contains when you instantiate one, and it is a requirement that you manually add steps e.g.
```
Procedure procedure = new Procedure();
procedure.addStep(step1);
procedure.addStep(step2);
...
```

Reply:
Totally agree, most of this is just to get something running enough to show myself that a chain of events is possible and instantiating classes is possible.  Note there are only print linesa dn no real logic in anything.  All of this needs to mature and be done correctly.  1) this is still exploratory and 2) I don't quite have all those skills yet, and 3) still have loads of questions around how to structure and do things